internal/indexer/indexer.go

```diff
@@ -723,6 +845,89 @@ func (i *Indexer) emitContainsForProject() {
 	}
 }
 
+func (i *Indexer) indexPackageDeclarations() {
+	i.visitEachPackage("Indexing package declarations", i.indexPackageDeclarationForPackage)
+}
+
+type DeclInfo struct {
+	HasDoc bool
+	Path   string
+}
+
+func (i *Indexer) indexPackageDeclarationForPackage(p *packages.Package) {
```

Comment:
Thank you for the inline comments in this function. They were very thoughtful and were helpful to navigate when unfamiliar with the code.